askama_parser/src/lib.rs

```diff
@@ -97,13 +113,21 @@ impl<'a> Ast<'a> {
         let (row, last_line) = source_before.lines().enumerate().last().unwrap_or_default();
         let column = last_line.chars().count();
 
+        let file_info = file_path.and_then(|file_path| {
+            let cwd = std::env::current_dir().ok()?;
+            Some((cwd, file_path))
+        });
         let msg = format!(
-            "{}problems parsing template source at row {}, column {} near:\n{}",
+            "{}problems parsing template source{} at row {}, column {} near:\n{}",
```

Comment:
So Rust messages look like this:

```rust
error: mismatched closing delimiter: `}`
   --> askama_derive/src/config.rs:95:13
    |
25  |     ) -> std::result::Result<Config<'a>, CompileError> {
    |                                                        - closing delimiter possibly meant for this
...
95  |         for (extensions, path in DEFAULT_ESCAPERS {
    |             ^ unclosed delimiter
...
106 |     }
    |     ^ mismatched closing delimiter
```

I propose that we make ours look more like:

```
error: failed to parse template source
  --> testing/templates/invalid_syntax.html:1:14
       "}"
 --> tests/ui/error_file_path.rs:3:10
  |
3 | #[derive(Template)]
  |          ^^^^^^^^
  |
  = note: this error originates in the derive macro `Template` (in Nightly builds, run with -Z macro-backtrace for more info)
```

(With exactly the second line appearing only if we have a source location here.)

Reply:
Looks better indeed. :+1: Makes a lot of diff though. ^^'

Reply:
Why does it make a lot of diff?

Reply:
Because we updated the message and all UI tests get a new output.